app/com/yetu/oauth2provider/oauth2/services/AuthorizeService.scala

```diff
@@ -35,7 +37,7 @@ class AuthorizeErrorHandler(clientService: IClientService,
     AuthorizeRequest(authorization.headers, authorization.params)
   }
 
-  def handleAuthorizeRequest[A](request: AuthorizeRequest, user: YetuUser): Either[OAuthError, AuthorizedClient] = try {
+  def handleAuthorizeRequest[A](request: AuthorizeRequest, user: YetuUser): Future[Either[OAuthError, AuthorizedClient]] = try {
```

Comment:
We opt to not do this now, because is it a complicated code there with nested Futures.. We will do this with you in person, then you can show us the best way.